### PR TITLE
Flatten LogicalExpression

### PR DESCRIPTION
**Pull request by dutor**
This refactor helps on the upcoming boolean expression analysis.

### CODE REVIEW

File: src/common/expression/LogicalExpression.h [hunk 1]
```diff
@@ -10,12 +10,19 @@
 #include "common/expression/BinaryExpression.h"
 
 namespace nebula {
-class LogicalExpression final : public BinaryExpression {
+class LogicalExpression final : public Expression {
```
Comment (darionyaphet) [inline src/common/expression/LogicalExpression.h:13]: why don't use Binary Expression ?

Comment (darionyaphet) [reply src/common/expression/LogicalExpression.h:13]: It looks like you want to overwrite the methods such as `left` and `right` ?

Comment (dutor) [reply src/common/expression/LogicalExpression.h:13]: Sorry about the confusion.
Yeah, I should have removed the _left_ and _right_ thing completely. Please wait a moment.

Comment (dutor) [reply src/common/expression/LogicalExpression.h:13]: Addressed

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**